### ISSUE TITLE
Path to Scripts Does Not Work For Tokenized Paths or For Paths Relative to Config File Path

### ISSUE THREAD

**Issue by DavidMtbg** [blipson89/Leprechaun#18]
I'm trying to use the NuGet package version and specify the path to the scripts which I want in source control (but not the binaries), but Leprechaun seems to always append whatever path I specify to the path where the executable is (<solution root>/packages/Leprechaun + <version number>/...) which is not to be in source control.

I've tried tokenized paths like $(ConfigDirectory), ./../../../Foundation/... and others, but it's always appended.

Here is the error output I'm getting since it can't find the scripts below the exe in the NuGet packages folder:
```C:\Users\david\Source\Repos\Website> cmd.exe /c gulp -b "C:\Users\david\Source\Repos\Website" --color --gulpfile "C:\Users\david\Source\Repos\Website\Gulpfile.js" 07-Code-Generate-Models
[16:21:36] Using gulpfile ~\Source\Repos\Website\Gulpfile.js
[16:21:36] Starting '07-Code-Generate-Models'...
  _                             _                       
 | |    ___ _ __  _ __ ___  ___| |__   __ _ _   _ _ __  
 | |   / _ \ '_ \| '__/ _ \/ __| '_ \ / _` | | | | '_ \ 
 | |__|  __/ |_) | | |  __/ (__| | | | (_| | |_| | | | |
 |_____\___| .__/|_|  \___|\___|_| |_|\__,_|\__,_|_| |_|
           |_|                       Version 1.0.1-pre01
                 @@                 .###@@::;%%&&00'
                @><@               .###@@::;%%&&00'
          ________)               .###@@::;%%&&00'
         |        |              .###@@::;%%&&00'
    _   _|===LI===|_            .###@@::;%%&&00'
   / \_(____________)          .###@@::;%%&&00'
   \  / (88 o  o 88)          .###@@::;%%&&00'
    \/\  88:  7 :88`         .###@@::;%%&&00'
     \/\ '88'=='88'         .###@@::;%%&&00'
      \ \__'8888'__________.###@@::;%%&&00'
       \___<\""/>_____/_/_.###@@::;%%&&00'
          /  ><  \       .###@@::;%%&&00'
        / __/--\__\     (oO@OoO@@o@oO@@o)
        '-.______.-'    /`"""""""""""""`\
          _|_||_|_     |                 |
       ___LI)||(LI___  |                 |
      (   ~~ || ~~   )  \               /
       `-----''-----`    '.___________.'
 
Loaded metadata for 17 configurations (0 total templates) in 156ms.
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.FileNotFoundException: Code generation script 'C:\Users\david\Source\Repos\Website\packages\Leprechaun.Console.1.0.1-pre01\lib\net46\.\src\Foundation\CodeGen\code\Scripts\Diagnostics.csx' was not found on disk.
   at Leprechaun.CodeGen.Roslyn.CSharpScriptCodeGenerator.<ProcessScripts>b__7_0(String script)
   at System.Linq.Parallel.SelectQueryOperator`2.SelectQueryOperatorResults.GetElement(Int32 index)
   at System.Linq.Parallel.QueryResults`1.get_Item(Int32 index)
   at System.Linq.Parallel.PartitionedDataSource`1.ListContiguousIndexRangeEnumerator.MoveNext(T& currentElement, Int32& currentKey)
   at System.Linq.Parallel.StopAndGoSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Linq.Parallel.QueryTask.<>c.<.cctor>b__10_0(Object o)
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Linq.Parallel.QueryTaskGroupState.QueryEnd(Boolean userInitiatedDispose)
   at System.Linq.Parallel.SpoolingTask.SpoolStopAndGo[TInputOutput,TIgnoreKey](QueryTaskGroupState groupState, PartitionedStream`2 partitions, SynchronousChannel`1[] channels, TaskScheduler taskScheduler)
   at System.Linq.Parallel.DefaultMergeHelper`2.System.Linq.Parallel.IMergeHelper<TInputOutput>.Execute()
   at System.Linq.Parallel.MergeExecutor`1.Execute()
   at System.Linq.Parallel.MergeExecutor`1.Execute[TKey](PartitionedStream`2 partitions, Boolean ignoreOutput, ParallelMergeOptions options, TaskScheduler taskScheduler, Boolean isOrdered, CancellationState cancellationState, Int32 queryId)
   at System.Linq.Parallel.PartitionedStreamMerger`1.Receive[TKey](PartitionedStream`2 partitionedStream)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.QueryOperator`1.ExecuteAndGetResultsAsArray()
   at System.Linq.ParallelEnumerable.ToArray[TSource](ParallelQuery`1 source)
   at Leprechaun.CodeGen.Roslyn.CSharpScriptCodeGenerator.ProcessScripts(String scripts)
   at lambda_method(Closure , Object[] )
   at Configy.Containers.MicroContainer.Activate(Type type, KeyValuePair`2[] unmappedConstructorParameters)
   at Configy.XmlContainerBuilder.<>c__DisplayClass5_0.<RegisterConfigTypeInterface>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Configy.Containers.MicroContainer.Resolve(Type type)
   at Configy.Containers.MicroContainer.Resolve[T]()
   at Leprechaun.Console.Program.<>c__DisplayClass0_0.<Main>b__0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Leprechaun.Console.Program.Main(String[] args)
[16:21:37] '07-Code-Generate-Models' errored after 529 ms
[16:21:37] Error: Command failed: C:\WINDOWS\system32\cmd.exe /s /c ".\packages\Leprechaun.Console.1.0.1-pre01\lib\net46\Leprechaun.console.exe /c .\src\Foundation\CodeGen\code\Leprechaun.config"
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.AggregateException: One or more errors occurred. ---> System.IO.FileNotFoundException: Code generation script 'C:\Users\david\Source\Repos\Website\packages\Leprechaun.Console.1.0.1-pre01\lib\net46\.\src\Foundation\CodeGen\code\Scripts\Diagnostics.csx' was not found on disk.
   at Leprechaun.CodeGen.Roslyn.CSharpScriptCodeGenerator.<ProcessScripts>b__7_0(String script)
   at System.Linq.Parallel.SelectQueryOperator`2.SelectQueryOperatorResults.GetElement(Int32 index)
   at System.Linq.Parallel.QueryResults`1.get_Item(Int32 index)
   at System.Linq.Parallel.PartitionedDataSource`1.ListContiguousIndexRangeEnumerator.MoveNext(T& currentElement, Int32& currentKey)
   at System.Linq.Parallel.StopAndGoSpoolingTask`2.SpoolingWork()
   at System.Linq.Parallel.SpoolingTaskBase.Work()
   at System.Linq.Parallel.QueryTask.BaseWork(Object unused)
   at System.Linq.Parallel.QueryTask.<>c.<.cctor>b__10_0(Object o)
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Linq.Parallel.QueryTaskGroupState.QueryEnd(Boolean userInitiatedDispose)
   at System.Linq.Parallel.SpoolingTask.SpoolStopAndGo[TInputOutput,TIgnoreKey](QueryTaskGroupState groupState, PartitionedStream`2 partitions, SynchronousChannel`1[] channels, TaskScheduler taskScheduler)
   at System.Linq.Parallel.DefaultMergeHelper`2.System.Linq.Parallel.IMergeHelper<TInputOutput>.Execute()
   at System.Linq.Parallel.MergeExecutor`1.Execute()
   at System.Linq.Parallel.MergeExecutor`1.Execute[TKey](PartitionedStream`2 partitions, Boolean ignoreOutput, ParallelMergeOptions options, TaskScheduler taskScheduler, Boolean isOrdered, CancellationState cancellationState, Int32 queryId)
   at System.Linq.Parallel.PartitionedStreamMerger`1.Receive[TKey](PartitionedStream`2 partitionedStream)
   at System.Linq.Parallel.UnaryQueryOperator`2.UnaryQueryOperatorResults.GivePartitionedStream(IPartitionedStreamRecipient`1 recipient)
   at System.Linq.Parallel.QueryOperator`1.ExecuteAndGetResultsAsArray()
   at System.Linq.ParallelEnumerable.ToArray[TSource](ParallelQuery`1 source)
   at Leprechaun.CodeGen.Roslyn.CSharpScriptCodeGenerator.ProcessScripts(String scripts)
   at lambda_method(Closure , Object[] )
   at Configy.Containers.MicroContainer.Activate(Type type, KeyValuePair`2[] unmappedConstructorParameters)
   at Configy.XmlContainerBuilder.<>c__DisplayClass5_0.<RegisterConfigTypeInterface>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Configy.Containers.MicroContainer.Resolve(Type type)
   at Configy.Containers.MicroContainer.Resolve[T]()
   at Leprechaun.Console.Program.<>c__DisplayClass0_0.<Main>b__0()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Leprechaun.Console.Program.Main(String[] args)
    at ChildProcess.exithandler (child_process.js:213:12)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:821:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
Process terminated with code 1.```

**Comment by blipson89** [comment]
Hi, 
`configDirectory` is case sensitive. Please try using `$(configDirectory)` which should resolve to where the config file is.

What arguments are you passing the `exe`? 

-Ben

**Comment by DavidMtbg** [comment]
I believe I have found the source of the issue (why it is neither detokenizing configDirectory nor starting from the configDirectory by default as I assumed all paths would do given [this comment in Leprechaun.config](https://github.com/blipson89/Leprechaun/blob/master/src/Leprechaun.Console/Leprechaun.config#L35)).  I think it is this block: https://github.com/blipson89/Leprechaun/blob/6def4aa96b56ffc686055861d7fe5309a4b7885b/src/Leprechaun.CodeGen.Roslyn/CSharpScriptCodeGenerator.cs#L97

`// not an absolute drive path, in which case we want to prepend the current exe path to it
					if (scriptPath[1] != ':')
					{
						scriptPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, scriptPath);
					}`

Perhaps it can be refactored to work similarly to this?:
https://github.com/blipson89/Leprechaun/blob/734c22799b63401df2bdab35477ebc2e1688ee61/src/Leprechaun/LeprechaunConfigurationBuilder.cs#L117

Thanks,
David

**Comment by blipson89** [comment]
we came to similar conclusions 😃  https://github.com/blipson89/Leprechaun/commit/ddf704dfe9116fc7f013eb534428ed112fb86f3e


Can you try this exe and see if that solves your problem? It'll ensure that all relative paths are converted to absolute, using the path to the exe as the base.
[Leprechaun-1.0.1-pre02.zip](https://github.com/blipson89/Leprechaun/files/2139208/Leprechaun-1.0.1-pre02.zip)

**Comment by blipson89** [comment]
I'll take a closer look tomorrow afternoon and get back to you. I agree - I think I misunderstood the issue you were having.

**Comment by DavidMtbg** [comment]
Per our outside conversation, I changed the way the path to the config was specified as you suggested (..\\..\\src\\Foundation\\CodeGen\\code\\Leprechaun.config instead of .\\src\\Foundation\\CodeGen\\code\\Leprechaun.config) and that resolved all issues with paths - thank you!  

Just a reminder that the version of the Runner package with the fix (1.0.1-pre02) is not on NuGet yet.

**Comment by blipson89** [comment]
This was merged in a long time ago. Closed.
